dietpi/dietpi-software

```diff
@@ -11615,6 +11655,71 @@ _EOF_
 			G_EXEC chown -R ffsync:ffsync /mnt/dietpi_userdata/firefox-sync
 
 		fi
+		
+                software_id=183 # Bitwarden RS
+                if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+                    Banner_Configuration
+
+                    Create_User -d /mnt/dietpi_userdata/bitwarden_rs bitwarden_rs
+
+                    # Make dir needed to run
+                    G_EXEC mkdir -p /mnt/dietpi_userdata/bitwarden_rs/
+
+                    # Make and edit the template config to have no web vault
+		    if [[ ! -f '/mnt/dietpi_userdata/bitwarden_rs/bitwarden_rs.env' ]]
+		    then
+			    G_EXEC cp /opt/bitwarden_rs/.env.template /mnt/dietpi_userdata/bitwarden_rs/bitwarden_rs.env
+			    G_CONFIG_INJECT 'WEB_VAULT_ENABLED=' 'WEB_VAULT_ENABLED=false' /mnt/dietpi_userdata/bitwarden_rs/bitwarden_rs.env
+			    G_CONFIG_INJECT 'DATA_FOLDER=' 'DATA_FOLDER=/mnt/dietpi_userdata/bitwarden_rs' /mnt/dietpi_userdata/bitwarden_rs/bitwarden_rs.env
+		    fi
+
+                    # Permissions
+                    G_EXEC chown -R bitwarden_rs:bitwarden_rs /mnt/dietpi_userdata/bitwarden_rs
+                    G_EXEC chmod +x /opt/bitwarden_rs/target/release/bitwarden_rs
+
+                    # Systemd service
+                    cat << _EOF_ > /etc/systemd/system/bitwarden_rs.service
+[Unit]
+Description=Bitwarden Server (Rust Edition)
+Documentation=https://github.com/dani-garcia/bitwarden_rs
+
+After=dietpi-boot.service network.target
+
+# Restart attempt only 5 times
+StartLimitIntervalSec=500
+StartLimitBurst=5
+
+[Service]
+# Server sometimes fails to start on startup, this should fix it
+Restart=on-failure
+RestartSec=5s
+# The user bitwarden_rs is run under. the working directory (see below) should allow write and read access to this user
+User=bitwarden_rs
+# The location of the .env file for configuration
+EnvironmentFile=/mnt/dietpi_userdata/bitwarden_rs/bitwarden_rs.env
+# The location of the compiled binary
+ExecStart=/opt/bitwarden_rs/target/release/bitwarden_rs
+# Set reasonable connection and process limits
+LimitNOFILE=1048576
+LimitNPROC=64
+# Isolate bitwarden_rs from the rest of the system
+PrivateTmp=true
+PrivateDevices=true
+ProtectHome=true
+ProtectSystem=strict
+# Only allow writes to the following directory and set it to the working directory (user and password data are stored here)
+WorkingDirectory=/mnt/dietpi_userdata/bitwarden_rs
+ReadWriteDirectories=/mnt/dietpi_userdata/bitwarden_rs
```

Comment:
This has been btw renamed to "ReadWritePaths" while the old name still works for backwards compatibility: https://github.com/caddyserver/caddy/pull/2798
https://www.freedesktop.org/software/systemd/man/systemd.exec.html#ReadWritePaths=